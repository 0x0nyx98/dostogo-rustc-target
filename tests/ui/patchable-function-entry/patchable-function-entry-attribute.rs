#![feature(patchable_function_entry)]
fn main() {}

#[patchable_function_entry(prefix_nops = 256, entry_nops = 0)]//~error: Expected integer value between 0 and 255.
pub fn too_high_pnops() {}

#[patchable_function_entry(prefix_nops = "stringvalue", entry_nops = 0)]//~error: Expected integer value between 0 and 255.
pub fn non_int_nop() {}

#[patchable_function_entry]//~error: malformed `patchable_function_entry` attribute input
pub fn malformed_attribute() {}

#[patchable_function_entry(prefix_nops = 10, something = 0)]//~error: Unexpected parameter name. Allowed names: prefix_nops, entry_nops
pub fn unexpected_parameter_name() {}

#[patchable_function_entry()]//~error: Must specify at least one parameter.
pub fn no_parameters_given() {}
