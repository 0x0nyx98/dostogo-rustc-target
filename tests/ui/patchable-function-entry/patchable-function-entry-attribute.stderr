error: malformed `patchable_function_entry` attribute input
  --> $DIR/patchable-function-entry-attribute.rs:10:1
   |
LL | #[patchable_function_entry]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: must be of the form: `#[patchable_function_entry(prefix_nops = m, entry_nops = n)]`

error: Expected integer value between 0 and 255.
  --> $DIR/patchable-function-entry-attribute.rs:4:42
   |
LL | #[patchable_function_entry(prefix_nops = 256, entry_nops = 0)]
   |                                          ^^^

error: Expected integer value between 0 and 255.
  --> $DIR/patchable-function-entry-attribute.rs:7:42
   |
LL | #[patchable_function_entry(prefix_nops = "stringvalue", entry_nops = 0)]
   |                                          ^^^^^^^^^^^^^

error: Unexpected parameter name. Allowed names: prefix_nops, entry_nops
  --> $DIR/patchable-function-entry-attribute.rs:13:46
   |
LL | #[patchable_function_entry(prefix_nops = 10, something = 0)]
   |                                              ^^^^^^^^^^^^^

error: Must specify at least one parameter.
  --> $DIR/patchable-function-entry-attribute.rs:16:1
   |
LL | #[patchable_function_entry()]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 5 previous errors

